File: javascript/src/notify/slack.ts
/**
 * Slack incoming-webhook notifier.
 *
 * The minimal accepted payload is `{ "text": "..." }`.
 */
import type { Message, Notifier } from './notify';
import { postJson } from './notify';

/** Posts to a Slack incoming webhook. */
export class SlackNotifier implements Notifier {
  constructor(private readonly url: string) {}

  /** Post the message as Slack mrkdwn. */
  async notify(m: Message): Promise<void> {
    await postJson(this.url, { text: slackText(m) });
  }
}

/** Render a Message as Slack mrkdwn. */
export function slackText(m: Message): string {
  const parts: string[] = [];
  if (m.title) {
    parts.push(`*${m.title}*`);
  }
  if (m.text) {
    parts.push(m.text);
  }
  if (m.link) {
    parts.push(`<${m.link}>`);
  }
  return parts.join('\n');
}

File: python/automation_agent/notify/slack.py
"""Slack incoming-webhook notifier.

The minimal accepted payload is ``{"text": "..."}``.
"""

from __future__ import annotations

from automation_agent.notify.notify import Message, post_json


class SlackNotifier:
    """Posts to a Slack incoming webhook."""

    def __init__(self, url: str) -> None:
        self.url = url

    def notify(self, m: Message) -> None:
        """Post the message as Slack mrkdwn."""
        post_json(self.url, {"text": slack_text(m)})


def slack_text(m: Message) -> str:
    """Render a Message as Slack mrkdwn."""
    parts: list[str] = []
    if m.title != "":
        parts.append(f"*{m.title}*")
    if m.text != "":
        parts.append(m.text)
    if m.link != "":
        parts.append(f"<{m.link}>")
    return "\n".join(parts)
